k9mail/src/main/java/com/fsck/k9/service/MailService.java

```diff
@@ -342,7 +344,10 @@ private void reschedulePushers(boolean hasConnectivity, boolean doBackground) {
 
     private void setupPushers() {
         boolean pushing = false;
-        for (Account account : Preferences.getPreferences(MailService.this).getAccounts()) {
+        List<Account> accounts =
+                new ArrayList<Account>(Preferences.getPreferences(MailService.this)
```

Comment:
And we are. Which makes me think this didn't fix it anyway and it's more subtle. I'll revert this for now and try and reproduce it later.